epifor/regions.py

```diff
@@ -3,11 +3,9 @@
 import logging
 import sys
 
-import yaml
+from .common import _fs, _n, yaml
```

Comment:
Pretty unclear names. Generally, I would prefer a more descriptive variable name (instead of `rs` or single digits). I am fine with more typing and longer lines compared to overhead with deciphering single char vars.

Reply:
Matter of style (`rs`), mildly agree on `_n`.

Reply:
@hnykda Important meta:

I appreciate where you come from, but the norms we even *should* try to endorse here are very different from something like long-term maintained, e.g. corporate projects. The goal is to save the world, not have nice code, and the returns on time+energy investments diminish quickly.

I had some intense discussions with (great) AI researchers who claimed that the tendency keeping your code nice is even a non-trivial minus in research - it focuses your energy and attention away from the goal, most code will be rewritten anyway (50% guess including this one), keeping code nice is an easy escape task from hard thinking (I know that too well!), and it seldom pays off beyond some basics. (E.g. I already spent >1 h here, more than half of those were +- relevant but still local stylistic issues.)

Good *code design* (modularity, narrow interfaces, delegation, somewhat extensible APIs) is another matter, though, and there the opinion mix is much more varied, although not animous.

(I respect your esthetics and value, though, and I know you are trying to be helpful with every comment.)

Reply:
> The goal is to save the world, not have nice code

As I wrote in the main review comment and many others, some of the notes are indeed details I don't really care about but were IMO worthy to be mentioned.

> I had some intense discussions with (great) AI researchers who claimed that the tendency keeping your code nice is even a non-trivial minus in research - it focuses your energy and attention away from the goal, most code will be rewritten anyway (50% guess including this one), keeping code nice is an easy escape task from hard thinking (I know that too well!), and it seldom pays off beyond some basics. (E.g. I already spent >1 h here, more than half of those were +- relevant but still local stylistic issues.)

I agree in general and see it as a tradeoff. I also strongly "believe" in pragmatism here and you can see from my github and OSS contributions I don't always thrive for "the best I could see". And I think that some things in this PR are on the good side of the tradeoff/pragmatic, but there are some which are IMHO on the wrong side of it. Some of the things I recommend are here to achieve the goal. Example: using a single letter variables without meaning is making it harder for _me_ and likely many others to contribute and actually _understand_ the code and its design, every reader has to spend time deciphering what it means (and this is write-once, read-many scenario). To understand the logic, my brain has to deal with translation hashmaps which are unnecessary. If I want to safe the world with this code, it's going to take more time. Compared to the cost you would have with using proper variable names which is almost free for you in today world with IDEs. Similar to absolute imports - I don't have to guess where a thing comes from, it gives me context. [Constants tells me what they stand for instead of magic numbers](https://github.com/epidemics/epifor/pull/10/files/4c2a25232038988d642c49d181016e8dc783e527#r397286037) - otherwise I have to search in the code it does what it should. Also, some of the things I mentioned are the language standards and take close to zero more effort (like docstrings - you hit the thing three times instead of once), making it consistent with whatever else people have seen and their IDEs helps them.

Apart from that, I believed that these could be useful for you as a programmer in other projects.

Also, as you know, I am basically running the https://github.com/epidemics/covid where 7+ people contribute. It's much smaller and it's a mess, it broke production because of some token for notion expiring. I hope it clearly demonstrates I am pragmatic in this and am trying to save world instead of bashing beginners with not using specific language pattern. It's also valuable datapoint based on which I am giving you these recommendations.